http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/MultiplexedChannelRecord.java

```diff
@@ -175,12 +176,20 @@ void release(Channel channel) {
         childChannels.remove(channel.id());
     }
 
-    public Future<Channel> getConnectionFuture() {
-        return connectionFuture;
+    boolean reusable() {
+        return !isPingInflight() && availableStreams.get() > 0 && !goAway;
```

Comment:
Why skip using it while a ping is in-flight?

Reply:
Because in the worst case, ping can take another 5 seconds to fail and the new requests picking up the same channel would end up getting "stuck" for 5 seconds. Establishing a new connection can prevent it from happening.